processor/transformprocessor/config.go

```diff
@@ -129,45 +132,72 @@ func (c *Config) Unmarshal(conf *confmap.Conf) error {
 
 var _ component.Config = (*Config)(nil)
 
+func (c *Config) ValidateTraceStatements(additionalSpanFuncs ...ottl.Factory[ottlspan.TransformContext]) error {
+	var errors error
+	pc, err := common.NewTraceParserCollection(component.TelemetrySettings{Logger: zap.NewNop()}, common.WithSpanParser(traces.SpanFunctions(additionalSpanFuncs)), common.WithSpanEventParser(traces.SpanEventFunctions()))
+	if err != nil {
+		return err
+	}
+	for _, cs := range c.TraceStatements {
+		_, err = pc.ParseContextStatements(cs)
+		if err != nil {
+			errors = multierr.Append(errors, err)
+		}
+	}
+	return errors
+}
+
+func (c *Config) ValidateLogStatements(additionalLogFuncs ...ottl.Factory[ottllog.TransformContext]) error {
+	var errors error
+	pc, err := common.NewLogParserCollection(component.TelemetrySettings{Logger: zap.NewNop()}, common.WithLogParser(logs.LogFunctions(additionalLogFuncs...)))
+	if err != nil {
+		return err
+	}
+	for _, cs := range c.LogStatements {
+		_, err = pc.ParseContextStatements(cs)
+		if err != nil {
+			errors = multierr.Append(errors, err)
+		}
+	}
+	return errors
+}
+
+func (c *Config) ValidateMetricStatements(additionalMetricFuncs ...ottl.Factory[ottlmetric.TransformContext]) error {
+	var errors error
+	pc, err := common.NewMetricParserCollection(component.TelemetrySettings{Logger: zap.NewNop()}, common.WithMetricParser(metrics.MetricFunctions(additionalMetricFuncs...)), common.WithDataPointParser(metrics.DataPointFunctions()))
+	if err != nil {
+		return err
+	}
+	for _, cs := range c.MetricStatements {
+		_, err := pc.ParseContextStatements(cs)
+		if err != nil {
+			errors = multierr.Append(errors, err)
+		}
+	}
+	return errors
+}
+
 func (c *Config) Validate() error {
 	var errors error
 
 	if len(c.TraceStatements) > 0 {
-		pc, err := common.NewTraceParserCollection(component.TelemetrySettings{Logger: zap.NewNop()}, common.WithSpanParser(traces.SpanFunctions()), common.WithSpanEventParser(traces.SpanEventFunctions()))
+		err := c.ValidateTraceStatements()
```

Comment:
Don't you need to pass the new functions as arguments to the new methods you've defined here?